volatility3/framework/plugins/windows/writecrashdump.py

```diff
@@ -0,0 +1,121 @@
+import math
+from typing import List, Type
+
+from volatility3.framework import interfaces, renderers, constants, exceptions
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins, configuration
+from volatility3.framework.layers import physical, intel
+from volatility3.framework.symbols import intermed
+from volatility3.plugins.windows import info
+
+
+class WriteCrashDump(plugins.PluginInterface):
+    """Runs the automagics and writes the output to a crashdump format file"""
+    default_block_size = 0x500000
+
+    _required_framework_version = (1, 0, 0)
+    _version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary', description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.IntRequirement(name = 'block_size',
+                                        description = "Size of blocks to copy over",
+                                        default = cls.default_block_size,
+                                        optional = True),
+            requirements.VersionRequirement("info", component = info.Info, version = (1, 0, 0))
+        ]
+
+    @classmethod
+    def write_crashdump(cls, context: interfaces.context.ContextInterface, layer_name: str, symbol_table: str,
+                        open_method: Type[interfaces.plugins.FileHandlerInterface]):
+        primary = context.layers[layer_name]
+        is_pae = isinstance(primary, intel.IntelPAE)
+        is_64_bit = isinstance(primary, intel.Intel32e)
+
+        if is_64_bit:
+            crashdump_json = 'crash64'
+            dump_header_name = '_DUMP_HEADER64'
+            valid_dump_suffix = [ord('6'), ord('4')]
+        else:
+            crashdump_json = 'crash'
+            dump_header_name = '_DUMP_HEADER'
+            valid_dump_suffix = [ord('M'), ord('P')]
+
+        config_path = 'whatever'
```

Comment:
Bad past @ikelos , very, very bad.  This sets a really bad example, and this needs changing before this gets committed.